src/singletons/WindowManager.cpp

```diff
@@ -622,4 +624,14 @@ void WindowManager::incGeneration()
     this->generation_++;
 }
 
+QJsonArray WindowManager::buildWindowArray(QString &settingsPath)
```

Comment:
Name could be changed to `loadWindowArray` to suggest that it does a disk operation.

I like that you pass the path as a reference but it needs to be changed to a `const QString&` since it should be immutable.

Reply:
you have to understand that I don't really have a clue how shit works in C++ 🤣 😂 👌